engine/runtime/src/main/scala/org/enso/compiler/pass/IRPass.scala

```diff
@@ -77,7 +77,8 @@ trait IRPass extends ProcessingPass {
     * @return the result of updating metadata in `copyOfIr` globally using
     *         information from `sourceIr`
     */
-  def updateMetadataInDuplicate[T <: IR](sourceIr: T, copyOfIr: T): T
+  def updateMetadataInDuplicate[T <: IR](@unused sourceIr: T, copyOfIr: T): T =
+    copyOfIr
```

Comment:
This introduces a substantial change that spans multiple Scala files. I have done it because in my own compiler pass, I wanted to provide the default implementation for this method, and I accidentally returned `sourceIr` instead of `copyOfIr`, which led to a bug that was very difficult to find.

To ensure that nobody has to bump into the same bug in the future, I provided the default implementation in this trait. Moreover, many of the inheritors provided the default implementation anyway.